web/api/v1/api.go

```diff
@@ -1121,7 +1121,12 @@ func (api *API) remoteRead(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 		for i, query := range req.Queries {
-			err := api.remoteReadQuery(ctx, query, externalLabels, func(set storage.SeriesSet) error {
+			err := api.remoteReadQuery(ctx, query, externalLabels, func(querier storage.Querier, selectParams *storage.SelectParams, filteredMatchers []*labels.Matcher) error {
+				// The streaming API provides sorted series.
+				set, _, err := querier.SelectSorted(selectParams, filteredMatchers...)
```

Comment:
Why is it only the streaming API that has it sorted?

Reply:
Thanos needs it. I don't see a reason to slow down the older endpoint, nor to incorrectly make it seem that it must be sorted.

Reply:
:+1:

Reply:
Yes, we rely on this heavily. Thanks!